src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -282,8 +284,9 @@ export class MeetingManager implements AudioVideoObserver {
     // This results into missing few meeting events triggered with audioVideoDidStop such as meetingEnded, meetingFailed and meetingStartFailed.
     // We may also loose audioInputUnselected and videoInputUnselected events as we choose null devices when an attendee leaves.
     // When a new AudioVideoFacade object is created remove and re-add the eventDidReceive observer which wont leak.
-    this.audioVideo.removeObserver({ eventDidReceive: this.eventDidReceive });
-    this.audioVideo.addObserver({ eventDidReceive: this.eventDidReceive });
+    this.eventDidReceiveObserver && this.audioVideo.removeObserver(this.eventDidReceiveObserver);
+    this.eventDidReceiveObserver = { eventDidReceive: this.eventDidReceive };
```

Comment:
Why cant we just initialize it in line 140 and just do remove it here?

Reply:
Yup that makes sense, update this.